Add allow/revoke ingress to EC2 security group action

RichieRunner opened the pull request:
Add the ability to test allowing/revoking an ingress rule to EC2 security group action

Signed-off by: RichieRunner <richie.ho@outlook.com>

RichieRunner commented:
@Lawouach ready to be reviewed 🙂

Lawouach commented:
Much appreciated @RichieRunner. For a simpler and better review, could you squash all your commits and force push?

Lawouach commented:
You may also have to rebase from mastre if you don't mind. Another PR went through.

RichieRunner commented:
> 
> 
> Much appreciated @RichieRunner. For a simpler and better review, could you squash all your commits and force push?

Done 👍

Lawouach commented on README.md:
```diff
@@ -368,7 +368,7 @@ def test_reboot_instance(aws_client):
 
 By using the [built-in Python module to mock objects][pymock], we can mock the
-EC2 client and assert we edo indeed call the appropriate method with the right
+EC2 client and assert that we do indeed call the appropriate method with the right
```
Appreciated :) Though it's usually better to have distinct PRs for these things :)

RichieRunner replied:
noted for next time 😄

Lawouach commented on chaosaws/ec2/actions.py:
```diff
@@ -657,3 +657,106 @@ def attach_instance_volume(client: boto3.client,
+            }
+        ]
+    }
+    if cidr_ip is not None:
+        request_kwargs['IpPermissions'][0]['IpRanges'][0]['CidrIp'] = cidr_ip  # noqa: E501
+    if ingress_security_group_id is not None:
+        request_kwargs['IpPermissions'][0]['UserIdGroupPairs'][0]['GroupId'] = ingress_security_group_id  # noqa: E501
```
Better to not disable the linter for these (it's okay on docstrings):

```
r = request_kwargs['IpPermissions'][0]
if cidr_ip is not None:
   r['IpRanges'][0]['CidrIp'] = cidr_ip
elif ingress_security_group_id is not None:
   r['UserIdGroupPairs'][0]['GroupId'] = ingress_security_group_id
```

RichieRunner replied:
thanks! i was struggling with how to resolve this 😓

Lawouach commented on chaosaws/ec2/actions.py:
```diff
@@ -657,3 +657,106 @@ def attach_instance_volume(client: boto3.client,
+        response = client.authorize_security_group_ingress(**request_kwargs)
+        return response
+    except ClientError as e:
+        raise FailedActivity(
```
Please use `ActivityFailed` as it replaced `FailedActivity` (the latter is deprecated and will be removed someday).

RichieRunner replied:
noted, done

Lawouach commented on chaosaws/ec2/actions.py:
```diff
@@ -657,3 +657,106 @@ def attach_instance_volume(client: boto3.client,
+            }
+        ]
+    }
+    if cidr_ip is not None:
+        request_kwargs['IpPermissions'][0]['IpRanges'][0]['CidrIp'] = cidr_ip  # noqa: E501
+    if ingress_security_group_id is not None:
+        request_kwargs['IpPermissions'][0]['UserIdGroupPairs'][0]['GroupId'] = ingress_security_group_id  # noqa: E501
+    try:
```
same comment as above

Lawouach commented on chaosaws/ec2/actions.py:
```diff
@@ -657,3 +657,106 @@ def attach_instance_volume(client: boto3.client,
+                e.response["Error"]["Message"]))
+
+
+def revoke_security_group_ingress(requested_security_group_id: str,
```
can we refactor the creation of the request kwargs into a dedicated function to avoid the duplication?

RichieRunner replied:
good idea, done

RichieRunner commented:
> 
> 
> Thanks. Looks like a good approach. A few things:
> 
>     * I would have rather prefered a rebasing, this would avoid the merge commit from master which always make the history log a bit of a pain
> 
>     * Could you add a change log entry?
> 
>     * Can you squash your commit messages into a single descriptive one?
> 
> 
> thanks :)

All resolved, squashed, rebased, and signed off 😌

Lawouach commented on CHANGELOG.md:
```diff
@@ -2,7 +2,15 @@
-[Unreleased]: https://github.com/chaostoolkit-incubator/chaostoolkit-aws/compare/0.13.0...HEAD
+[Unreleased]: https://github.com/chaostoolkit-incubator/chaostoolkit-aws/compare/0.14.0...HEAD
+
+## [0.14.0][]
```
for next time, no need to set the version in the PR. We do this on release :)

that's okay for this one :)
